presto-hive/src/main/java/com/facebook/presto/hive/metastore/CachingHiveMetastore.java

```diff
@@ -844,15 +844,23 @@ private void invalidatePartitionCache(String databaseName, String tableName)
     }
 
     private Set<String> loadRoles(String user)
+            throws Exception
     {
-        try (HiveMetastoreClient client = clientProvider.createMetastoreClient()) {
-            List<Role> roles = client.listRoles(user, USER);
-            if (roles == null) {
-                return ImmutableSet.of();
-            }
-            return ImmutableSet.copyOf(roles.stream()
-                    .map(Role::getRoleName)
-                    .collect(toSet()));
+        try {
+            return retry()
+                    .stopOn(MetaException.class)
```

Comment:
Every metastore method throws `MetaException` but for most things we treat it as transient (such as in `loadAllTables`) because it can be thrown for transient reasons like a database error on the metastore side.

So remove this `stopOn()`.